service/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java

```diff
@@ -63,33 +87,43 @@ class WorkspaceServiceTest extends BaseConnectedTest {
           .token(Optional.of("fake-token"))
           .email("fake@email.com")
           .subjectId("fakeID123");
+  public static final String SPEND_PROFILE_ID = "wm-default-spend-profile";
 
-  @Autowired private WorkspaceService workspaceService;
-  @Autowired private JobService jobService;
+  @MockBean private DataRepoService mockDataRepoService;
+  /** Mock SamService does nothing for all calls that would throw if unauthorized. */
+  @MockBean private SamService mockSamService;
+
+  @Autowired private ControlledResourceService controlledResourceService;
   @Autowired private CrlService crl;
-  @Autowired private SpendConnectedTestUtils spendUtils;
+  @Autowired private GcpCloudContextService gcpCloudContextService;
+  @Autowired private JobService jobService;
   @Autowired private ReferencedResourceService referenceResourceService;
   @Autowired private ResourceDao resourceDao;
+  @Autowired private SpendConnectedTestUtils spendUtils;
   @Autowired private WorkspaceConnectedTestUtils testUtils;
-  @MockBean private DataRepoService dataRepoService;
-  /** Mock SamService does nothing for all calls that would throw if unauthorized. */
-  @MockBean private SamService mockSamService;
+  @Autowired private WorkspaceService workspaceService;
 
   @BeforeEach
   void setup() throws Exception {
-    doReturn(true).when(dataRepoService).snapshotReadable(any(), any(), any());
+    doReturn(true).when(mockDataRepoService).snapshotReadable(any(), any(), any());
     // By default, allow all spend link calls as authorized. (All other isAuthorized calls return
-    // false by Mockito default.
+    // false by Mockito default).
     Mockito.when(
             mockSamService.isAuthorized(
                 Mockito.any(),
-                Mockito.eq(SamResource.SPEND_PROFILE),
+                eq(SamResource.SPEND_PROFILE),
                 Mockito.any(),
-                Mockito.eq(SamSpendProfileAction.LINK)))
+                eq(SamSpendProfileAction.LINK)))
         .thenReturn(true);
+    final String policyGroup = "terra-workspace-manager-test-group@googlegroups.com";
     // Return a valid google group for cloud sync, as Google validates groups added to GCP projects.
     Mockito.when(mockSamService.syncWorkspacePolicy(any(), any(), any()))
-        .thenReturn("terra-workspace-manager-test-group@googlegroups.com");
+        .thenReturn(policyGroup);
+
+    doReturn(policyGroup).when(mockSamService).syncResourcePolicy(
```

Comment:
This seems redundant to the previous mock, just with different syntax. It takes `any()` which I understand to be a superset of the `any(ControlledResource.class)` you have here.

Reply:
The previous mock is for the function `syncWorkspacePolicy()` and this one is for `syncResourcePolicy()`. Using a naked `any()` makes me feel icky, like I'm writing JavaScript.

Reply:
Ah, I missed that. Thanks for explaining!